pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -894,7 +1040,16 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 			activated = true
 		}
 	}
-	p.moveRequestCycle = p.schedulingCycle
+
+	// AddUnschedulableIfNotPresent might get called for in-flight Pods later, and in
```

Comment:
if `p.InFlightPods` is empty, we should not store the event. There is no point, and nothing would clean those entries.

Reply:
Good catch....